src/Transport/Topology/MetaModel/NamespacesDefinition.cs

```diff
@@ -0,0 +1,69 @@
+﻿namespace NServiceBus.AzureServiceBus
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    public class NamespacesDefinition : IEnumerable<NamespaceDefinition>
+    {
+        private static readonly string DefaultName = "default";
+        private static readonly string Prefix = "namespace-";
+
+        private readonly List<NamespaceDefinition> _inner;
+
+        public NamespacesDefinition()
+        {
+            _inner = new List<NamespaceDefinition>();
+        }
+
+        public int Count => _inner.Count;
+
+        public void Add(string name, string connectionString)
+        {
+            var definition = new NamespaceDefinition(name, connectionString);
+            if (_inner.Contains(definition)) return;
+
+            var defaultDefinition = _inner.SingleOrDefault(x => x.Name == DefaultName && x.ConnectionString == connectionString);
+            _inner.Remove(defaultDefinition);
+            _inner.Add(definition);
```

Comment:
Got it, thank you.
Does that mean something like this would take place? 
```c#
config.UseTransport<ASB>()
   .ConnectionString("defaultconn")
   .NamespacePartitioning().AddNamespace("defaultconn");
```
Should we throw an exception saying that there's a conflict between default and named namespace. Since it's configuration time, owner of the configuration should only register it once. Thoughts?